### DISCUSSION TITLE
Block all unmocked requests

### DISCUSSION THREAD

**Discussion by amd-9** [mswjs/msw#874] [Q&A] [answered]
Is there any way to block all unmocked request for whole testing environment?

**Comment by kettanaito** [comment, marked as answer]
Hey, @amd-9.

Yes, you can set the `onUnhandledRequest` option to "error". This will result in an error whenever an unhandled request happens, failing your test:

```js
server.listen({ onUnhandledRequest: 'error' })
```

[See the docs](https://mswjs.io/docs/api/setup-server/listen#onunhandledrequest).

**Reply by amd-9** [reply]
@kettanaito thanks! Seems like it works only on per test basis. I thought there must be some way to configure this behaviour for all test suites like we do for common config in `setupTests.js` for jest

**Reply by kettanaito** [reply]
If you configure the `server` in `setupTest.js` it will affect all requests in all tests.

**Reply by amd-9** [reply]
To make things clear. For every component I've got a `[ComponentName].test.ts` file. Each file contains section

```
const server = setupServer(
 // mock API endpoints setup for exact components
);

beforeAll(() => server.listen({ onUnhandledRequest: 'error' }));
afterEach(() => server.resetHandlers());
afterAll(() => server.close());
```

If I'll configure `server.listen({ onUnhandledRequest: 'error' }))` in `setupTest.js` I'll have to configure all API mock in `setupServer` for all my components in `setupTest.js` also?

**Reply by kettanaito** [reply]
If you're using MSW consistently in multiple test files move its setup to the global testing setup (`setupTest.js` in your case), and have it there _only once_. There is no need to re-instantiate `setupServer` in this case.

**Reply by amd-9** [reply]
Thanks for help!

**Reply by amd-9** [reply]
@kettanaito Sometimes API is very big (lot's of endpoints) that makes global setup in (`setupTests.js` for example) overwhelming and hard to navigate through or specific test requires specific data in mocked responses. I understand that there is no "silver bullet". But may be there is some recommendation to consider for such scenarios of MSW usage?

**Reply by kettanaito** [reply]
Remember that request handlers are regular functions. You can move them to another directory to lift some weight from the global setup.

What I'd recommend:

```js
// setupTest.js
import { user, checkout, contacts } from './handlers'

setupSerer(
  ...user,
  ...checkout,
  ...contacts
)
```

**Reply by amd-9** [reply]
Many thanks!
